src/toil_scripts/batch_alignment/bwa_alignment.py

```diff
@@ -1,223 +1,254 @@
+from toil_scripts.tools.preprocessing import run_samtools_faidx, run_bwa_index
 
-_log = logging.getLogger(__name__)
 
-def download_shared_files(job, inputs, sample, output_dir, rg_line=None):
+def download_reference_files(job, inputs, samples):
     """
     Downloads shared files that are used by all samples for alignment
 
     :param JobFunctionWrappingJob job: passed automatically by Toil
-    :param JobFunctionWrappingJob job: Passed by Toil automatically
     :param Namespace inputs: Input arguments (see main)
+    :param list[list[str, list]] samples: Sample UUIDs and URLs
     """
-    inputs.sample = sample
-    inputs.rg_line = rg_line
-    inputs.output_dir = output_dir
-    job.fileStore.logToMaster('Downloading shared files for alignment.')
-
-    shared_files = [inputs.ref, inputs.amb, inputs.ann, inputs.bwt, inputs.pac, inputs.sa, inputs.fai]
-    if getattr(inputs, 'alt', None):
-        shared_files.append(inputs.alt)
-
-    shared_ids = []
-    for url in shared_files:
-        shared_ids.append(job.addChildJobFn(download_url_job, url).rv())
-    job.addFollowOnJobFn(parse_config, shared_ids, inputs)
-
-
-def parse_config(job, shared_ids, inputs):
-    """
-    Stores the UUID and urls associated with the input files to be retrieved.
-    Configuration file has one sample per line, with the following format:  UUID\t1st_url\t2nd_url
-
-    :param JobFunctionWrappingJob job: Passed by Toil automatically
-    :param dict shared_ids: stores FileStoreIDs
-    :param Namespace inputs: Input arguments
-    """
-    if inputs.sample:
-        uuid, url1, url2 = inputs.sample
-        mock_bam = '.'.join(url1.split('.')[:-2])[:-2] + ".bam"
-        samples = [[uuid, [url1, url2], mock_bam]]
-    else:
-        samples = []
-
-        with open(inputs.manifest, 'r') as f_in:
-            for line in f_in:
-                if not line.isspace() and not line.startswith('#'):
-                    line = line.strip().split('\t')
-                    assert len(line) == 3, 'Improper formatting. Expected UUID\tURl1\tURL2. Received: {}'.format(line)
-                    uuid = line[0]
-                    urls = line[1:]
-                    mock_bam = '.'.join(line[1].split('.')[:-2])[:-2] + ".bam"
-                    samples.append((uuid, urls, mock_bam))
-    inputs.maxCores = int(inputs.maxCores) if hasattr(inputs, "maxCores") else sys.maxint
     inputs.cores = min(inputs.maxCores, multiprocessing.cpu_count())
-    job.fileStore.logToMaster('Parsed configuration file.')
-    job.addChildJobFn(map_job, download_sample, samples, inputs, shared_ids, cores=1, disk=inputs.file_size)
-
-
-def download_sample(job, sample, inputs, ids):
+    shared_ids = {}
+    urls = [('amb', inputs.amb), ('ann', inputs.ann), ('bwt', inputs.bwt),
+            ('pac', inputs.pac), ('sa', inputs.sa)]
+    # If all inputs provided, download from S3
+    download_ref = job.wrapJobFn(download_url_job, inputs.ref, disk='3G')
```

Comment:
> This reminds me of the famous paper "Hardcoding resource requirements considered harmful."

Agreed, but I think it's a bit overkill to ask the user for the reference size in the config...  I don't mind bumping this up to something like 5G or 10G to be on the safe side, but I've personally never used a reference bigger than hg19.fa / hg38.fa.